### PR TITLE
WIP cli.yml

### PR DESCRIPTION
**Pull request by mwritter**
### Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

#### Type of Change

Please delete options that are not relevant (including this descriptive text).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

### Checklist: (Feel free to delete this section upon completion)

- [ ] I have included a [changeset](../CONTRIBUTING.md#changesets) if this change will require a version change to one of the packages.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My code follows the style guidelines of this project (I have run `yarn format`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes (I have run `yarn test`)
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        run: pnpm run build:cli
+
+      - name: Install spectacle
+        run: pnpm install spectacle
+
+      - name: create examples directory and move into it
+        run: mkdir cli-build-examples
+
+      - name: ${{ matrix.output-types }} example
+        run: spectacle -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
```
Comment (ryan-roemer) [inline .github/workflows/cli.yml:54]: Do we also want to:

- Install the new project
- Build the deck

?

Comment (mwritter) [reply .github/workflows/cli.yml:54]: This is definitely a wip 😅, I believe this command creates the new example project, but yes after my talk with @gksander Im sure we want to `install` and `start` and `test` the example projects. Im just trying to get this to run right now tbh - 🤔 Maybe the github action will still be picked up in a 'draft pr'

Comment (ryan-roemer) [reply .github/workflows/cli.yml:51]: Sorry to jump the gun on a WIP request. Good luck and carry-on!

Comment (mwritter) [reply .github/workflows/cli.yml]: 👍 - Yep looks like the new `workflows/cli.yml` does get pick up when in draft pr so Ill keep it here until its working!

File: .github/workflows/cli.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        run: pnpm install
+
+      - name: Build CLI
+        run: pnpm run build:create-spectacle
+
+      - name: Install ts-node
+        run: sudo npm install -g ts-node
+
+      - name: Create ${{ matrix.output-types }} example project
+        run: ts-node ./packages/create-spectacle/src/cli.ts -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
```
Comment (ryan-roemer) [inline .github/workflows/cli.yml:56]: The build creates a usable JS project, so we can just use:

```suggestion
      - name: Create ${{ matrix.output-types }} example project
        run: node ./packages/create-spectacle/bin/cli.js -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
```

File: .github/workflows/cli.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build CLI
+        run: pnpm run build:create-spectacle
+
+      - name: Install ts-node
```
Comment (ryan-roemer) [inline .github/workflows/cli.yml:52]: The build creates a usable JS project, so we can just use:

```suggestion
      - name: Create ${{ matrix.output-types }} example project
        run: node ./packages/create-spectacle/bin/cli.js -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
```

File: .github/workflows/cli.yml (hunk 2 above)
Comment (ryan-roemer) [inline .github/workflows/cli.yml:56]: The build creates a usable JS project, so we can just use:

```suggestion
      - name: Create ${{ matrix.output-types }} example project
        run: node ./packages/create-spectacle/bin/cli.js -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
```

File: .github/workflows/cli.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        run: sudo npm install -g ts-node
+
+      - name: Create ${{ matrix.output-types }} example project
+        run: ts-node ./packages/create-spectacle/src/cli.ts -t ${{ matrix.output-types }} -n example-${{ matrix.output-types }}
+
+      # cant cd into onepage its just a html file
+      - name: Start project
+        run: |
```
Comment (ryan-roemer) [inline .github/workflows/cli.yml:60]: If easier, you can specify `run.working-directory` fields to avoid needing repeated `--prefix` flags.

Comment (mwritter) [reply .github/workflows/cli.yml:60]: Nice 👍

**Comment by gksander** [comment]
Closing since this has been absorbed into a merged PR.

**Pull request closed without merging**